lab/src/raft/request_vote.go

```diff
@@ -0,0 +1,86 @@
+package raft
+
+//
+// example RequestVote RPC arguments structure.
+// field names must start with capital letters!
+//
+type RequestVoteArgs struct {
+	// Your data here (2A, 2B).
+	Term int
+	CandidateId int
+	LastLogIndex int
+	LastLogTerm int
+}
+
+//
+// example RequestVote RPC reply structure.
+// field names must start with capital letters!
+//
+type RequestVoteReply struct {
+	// Your data here (2A).
+	Term int
+	VoteGranted bool
+}
+
+//
+// example RequestVote RPC handler.
+//
+func (rf *Raft) RequestVote(args *RequestVoteArgs, reply *RequestVoteReply) {
+	// Your code here (2A, 2B).
+	rf.mu.Lock()
+
+	localLastLogIndex := len(rf.log)
```

Comment:
Should `localLastLogIndex` be `len(rf.log) - 1` instead? Since you're comparing it with `args.LastLogIndex` on line 40 and they won't be equal if the local log size is the same as the peer's.

Reply:
Note that in paper states that first index is 1, thus the index's value is actually the length of log